trade-finance-manager-api/src/graphql/reducers/mappings/deal/mapTotals.js

```diff
@@ -1,11 +1,24 @@
 const { formattedNumber } = require('../../../../utils/number');
+const { calculateNewFacilityValue, calculateAmendmentTotalExposure, isValidCompletedValueAmendment } = require('../../../helpers/amendment.helpers');
+const api = require('../../../../v1/api');
+const { CURRENCY } = require('../../../../constants/currency.constant');
 
-const mapTotals = (facilities) => {
+const mapTotals = async (facilities) => {
   const totals = {};
 
   // total value of all facilities
-  const facilitiesValue = facilities.map((facility) => {
-    const { facilitySnapshot, tfm } = facility;
+  const facilitiesValue = await Promise.all(facilities.map(async (facility) => {
+    const { facilitySnapshot, tfm, _id } = facility;
+
```

Comment:
At any point throughout the call stack are we ensuring properties de-structured here does exists before making any API calls, in case they don't and if no checks are in place then will cause service to crash. If check is in place then please ignore.

Reply:
done